services/ecr/config/samples/full-example-repo.yaml

```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: "ecr.services.k8s.aws/v1alpha1"
+kind: Repository
+metadata:
+    name: "test-full-repository-from-ack"
+spec:
+    repositoryName: "test-full-repository-from-ack"
+    encryptionConfiguration:
+        encryptionType: KMS
+        # This key needs to exist already
+        kmsKey: "9d319c68-60ea-4ed6-8a33-2e09086fad2d"
+    imageScanningConfiguration:
+        scanOnPush: true
+    imageTagMutability: true
```

Comment:
When I apply this manifest, I'm getting the following failure:

```
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ kubectl apply -f services/ecr/config/samples/full-example-repo.yaml
The Repository "test-full-repository-from-ack" is invalid: spec.imageTagMutability: Invalid value: "boolean": spec.imageTagMutability in body must be of type string: "boolean"
```

That's because this field isn't a boolean :) It's actually a string enum, with valid values of "IMMUTABLE" and "MUTABLE".

Replacing `true` with `"true"` results in the CR being created successfully (since we don't have validating webhook yet...) but the controller logs errors like this:

```
2020-08-14T12:27:55.404Z	ERROR	controller-runtime.controller	Reconciler error	{"controller": "repository", "request": "default/test-full-repository-from-ack", "error": "InvalidParameterException: Invalid parameter at 'imageTagMutability' failed to satisfy constraint: 'Member must satisfy enum value set: [IMMUTABLE, MUTABLE]'"}
```

Changing the value to "IMMUTABLE" results in successful reconcilation.

Reply:
`scanOnPush` is a boolean (unquoted) but `imageTagMutability` needs quotes? That seems like it would be a bad experience for users who don't know the code.

Should I change the examples and add a comment saying it should be quoted or do we want to change the type?

Reply:
imageTagMutability isn't a boolean, and "true" isn't a valid value. Only "IMMUTABLE" and "MUTABLE" are valid values.